openedx/features/sketchfab_viewer/static/js/ModelViewer/Components/viewer.js

```diff
@@ -0,0 +1,263 @@
+                  for(let i=0; i < results.results.length; i++){
+                      temp.push([results.results[i].name, results.results[i].thumbnails.images[1].url, results.results[i].uid])           
+                  }
+                  this.setState({ modelsReturned: temp, previous_page_url: results.previous, next_page_url: results.next});
+              }
+        )
+      }
+    }
+
+   searchAPI(){
+        return new Promise((resolve, reject) => {
+            axios.get(`https://api.sketchfab.com/v3/search?type=models&q=${this.state.searchName}&count=6`,
+            {headers: {'Authorization' : 'Token d1a2be40a3034b3794fe540de52f5c11'}})
+                .then(resp => resolve(resp.data))
+                .catch(resp => alert(resp));
+        });
+    }
+
+    search(){          
+      console.log("searching")                             //call API fxn, then assign into temp icons and names, then save in state
+        this.searchAPI().then(
+            results =>
+            {
+                var temp = []
+                for(let i=0; i < results.results.length; i++){
+                    temp.push([results.results[i].name, results.results[i].thumbnails.images[1].url, results.results[i].uid])           
+                }
+                this.setState({ modelsReturned: temp, previous_page_url: results.previous, next_page_url: results.next});
+            }
+        )
+    }
+
+
+
+    handleChange = event => {
+        this.setState({
+            searchName: event.target.value
+        });
+    }
+
+    render(){
+        return(
+<div>
+  <head>
+    <title>Sketchfab Viewer API example</title>
+  </head>
+  <body>
+    <div>
+      <Navbar
+        variant="light"
+        sticky="top"
+        style={{
+          borderStyle: "solid",
+          borderColor: "white white red white"
+        }}
+      >
+        <Navbar.Brand href="#home">
+          <img
+            alt=""
+            src="https://d32r2qmogsjfiy.cloudfront.net/2020/03/Redhouse-logo.png"
+            width="250"
+            className="d-inline-block align-top"
+            style={{ marginLeft: "0.5em" }}
+          />
+        </Navbar.Brand>
+        <Nav className="ml-auto">
+          <Nav.Link href="/login">
+            <Button variant="primary">Register</Button>
+          </Nav.Link>
+        </Nav>
+      </Navbar> 
+    </div>
+    <div>
+      <img
+        src="https://static.stambol.com/wordpress/wp-content/uploads/2018/05/ARVRHeadsetDesign_1640x894.jpg"
+        style={{ width: "100%", height: "50%" }}
+      />
+    </div>
+
+    <div class="container-flex p-5">
+      <center>
+        <h1 class="m-3">Model Viewer</h1>
+      </center>
+      <h4>Search for images to display in Virtual Reality:</h4>
+
+      
+        <div class="col-6 d-flex justify-content-end">
+          <input
+            type="text"
+            class="form-control"
+            placeholder="e.g. Car"
+            style={{
+              

```

Comment:
empty styles

Reply:
Code is provided by the client themselves.